src/views/Home/components/Banners/hooks/useMultipleBannerConfig.tsx

```diff
@@ -20,24 +20,27 @@ import useIsRenderLotteryBanner from './useIsRenderLotteryBanner'
 export const useMultipleBannerConfig = () => {
   const isRenderIFOBanner = useIsRenderIfoBanner()
   const isRenderLotteryBanner = useIsRenderLotteryBanner()
-  return useMemo(
-    () =>
-      [
-        {
-          shouldRender: isRenderIFOBanner,
-          banner: <IFOBanner />,
-        },
-        {
-          shouldRender: isRenderLotteryBanner,
-          banner: <LotteryBanner />,
-        },
-        {
-          shouldRender: false,
-          banner: <CompetitionBanner />,
-        },
-      ]
-        .filter((d) => d.shouldRender)
-        .map((d) => d.banner),
-    [isRenderIFOBanner, isRenderLotteryBanner],
-  )
+  return {
+    bannerList: useMemo(
+      () =>
+        [
+          {
+            shouldRender: isRenderIFOBanner,
+            banner: <IFOBanner />,
+          },
+          {
+            shouldRender: isRenderLotteryBanner,
+            banner: <LotteryBanner />,
+          },
+          {
+            shouldRender: false,
+            banner: <CompetitionBanner />,
+          },
+        ]
+          .filter((d) => d.shouldRender)
+          .map((d) => d.banner),
+      [isRenderIFOBanner, isRenderLotteryBanner],
+    ),
+    isLoading: !isRenderLotteryBanner,
```

Comment:
i think all components can return {isRender, isLoading} object instead of just boolean then here we can check that if any isLoading true we can mark this as true. Wdyt? In that way we will not have dependency to lottery here

Reply:
Yeah, that's true.

Reply:
Yes, but still depending on lottery loading

Reply:
added IFO loading